src/EconomicStrategy/EconomicStrategyManager.ts

```diff
@@ -83,65 +86,32 @@ export class EconomicStrategyManager {
     return EconomicStrategyStatus.CLAIM;
   }
 
-  /**
-   * Calculates the correct gas price to use for execution, taking into consideration
-   * the economicStrategy `maxGasSubsidy` and the current network conditions.
-   * @param {TransactionRequest} txRequest Transaction Request object to check.
-   * @param {Config} config Configuration object.
-   */
   public async getExecutionGasPrice(txRequest: ITxRequest): Promise<BigNumber> {
     const { average } = await this.util.getAdvancedNetworkGasPrice();
-
-    let currentNetworkPrice = average;
-
-    if (!this.strategy) {
-      return currentNetworkPrice;
-    }
-
-    if (this.strategy.usingSmartGasEstimation) {
-      const smartGasEstimate = await this.smartGasEstimation(txRequest);
-      if (smartGasEstimate) {
-        currentNetworkPrice = smartGasEstimate;
-      }
-    }
-
-    const { maxGasSubsidy } = this.strategy;
-
-    if (typeof maxGasSubsidy !== 'undefined' && maxGasSubsidy !== null) {
-      const minGasPrice = txRequest.gasPrice;
-      const maxGasPrice = minGasPrice.plus(minGasPrice.times(maxGasSubsidy / 100));
-
-      if (currentNetworkPrice.lessThan(minGasPrice)) {
-        return minGasPrice;
-      } else if (
-        currentNetworkPrice.greaterThanOrEqualTo(minGasPrice) &&
-        currentNetworkPrice.lessThan(maxGasPrice)
-      ) {
-        return currentNetworkPrice;
-      } else if (currentNetworkPrice.greaterThanOrEqualTo(maxGasPrice)) {
-        return maxGasPrice;
-      }
-    }
-
-    return currentNetworkPrice;
+    const currentNetworkPrice = this.strategy.usingSmartGasEstimation
```

Comment:
I think it can be simplified to:
`
const currentNetworkPrice = (this.strategy.usingSmartGasEstimation && await this.smartGasEstimation(txRequest)) || average
`

This way there's no need to repeat `average` twice.